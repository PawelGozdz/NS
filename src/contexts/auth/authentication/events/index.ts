export * from './on-user-updated/on-user-updated.handler';
