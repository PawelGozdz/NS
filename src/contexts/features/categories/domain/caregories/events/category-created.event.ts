import { Event } from '@libs/cqrs';

export class CategoryCreatedEvent extends Event<CategoryCreatedEvent> {
  id: number;

  name: string;

  description?: string | null;

  parentId?: number | null;

  constructor(event: CategoryCreatedEvent) {
    super(event);

    Object.assign(this, event);
  }
}
