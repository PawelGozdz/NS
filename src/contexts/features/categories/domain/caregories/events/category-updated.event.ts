import { Event } from '@libs/cqrs';

export class CategoryUpdatedEvent extends Event<CategoryUpdatedEvent> {
	id: number;
	name: string;
	description: string | null;
	parentId: number | null;
	context: string;

	constructor(event: CategoryUpdatedEvent) {
		super(event);

		Object.assign(this, event);
	}
}
