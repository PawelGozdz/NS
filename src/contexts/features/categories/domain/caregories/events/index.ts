export * from './category-created.event';
export * from './category-updated.event';
