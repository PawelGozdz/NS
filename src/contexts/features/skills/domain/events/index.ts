export * from './skill-created.event';
