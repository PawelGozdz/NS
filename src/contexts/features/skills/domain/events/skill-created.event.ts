import { Event } from '@libs/cqrs';

export class SkillCreatedEvent extends Event<SkillCreatedEvent> {
  id: number;

  name: string;

  description?: string | null;

  parentId?: number | null;

  context: string;

  constructor(event: SkillCreatedEvent) {
    super(event);

    Object.assign(this, event);
  }
}
