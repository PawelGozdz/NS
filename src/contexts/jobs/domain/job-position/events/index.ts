export * from './job-position-created.event';
export * from './job-position-updated.event';
