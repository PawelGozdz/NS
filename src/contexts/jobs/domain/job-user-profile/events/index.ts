export * from './job-user-profile-created.event';
export * from './job-user-profile-updated.event';
