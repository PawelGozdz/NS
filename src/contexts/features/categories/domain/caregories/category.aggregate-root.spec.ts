import { Category } from './category.aggregate-root';
import { CategoryAggregateRootFixtureFactory } from './category.aggregate-root.fixture';
import { CategoryCreatedEvent, CategoryUpdatedEvent } from './events';

describe('Category', () => {
	afterEach(() => {
		jest.resetModules();
		jest.restoreAllMocks();
	});

	const category = CategoryAggregateRootFixtureFactory.create();

	describe('create', () => {
		describe('Success', () => {
			it('should apply CategoryCreatedEvent with property values and no ID', () => {
				// arrange

				// act
				const entity = Category.create(category);

				// assert
				const events = entity.getUncommittedEvents();

				expect(events.length).toEqual(1);
				expect(events[0]).toBeInstanceOf(CategoryCreatedEvent);

				expect(events[0]).toMatchSnapshot({
					id: undefined,
					name: category.name,
					description: category.description,
					parentId: category.parentId,
					context: category.context,
				});
			});
		});

		describe('Success', () => {
			it('should apply CategoryCreatedEvent with property values and ID', () => {
				// arrange

				// act
				const entity = Category.create({
					...category,
					id: 1,
				});

				// assert
				const events = entity.getUncommittedEvents();

				expect(events.length).toEqual(1);
				expect(events[0]).toBeInstanceOf(CategoryCreatedEvent);

				expect(events[0]).toMatchSnapshot({
					id: expect.any(Number),
					name: category.name,
					description: category.description,
					parentId: category.parentId,
					context: category.context,
				});
			});
		});
	});

	describe('update', () => {
		it('should apply CategoryUpdatedEvent with properties to be updated', () => {
			// arrange
			const newContext = 'new-context';

			// act
			category.update({ context: newContext });

			// assert
			const events = category.getUncommittedEvents();

			expect(events[0]).toBeInstanceOf(CategoryUpdatedEvent);
			expect(events[0]).toMatchSnapshot();
		});
	});
});
