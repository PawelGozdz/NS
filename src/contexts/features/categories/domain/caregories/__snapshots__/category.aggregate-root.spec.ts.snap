// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Category create Success should apply CategoryCreatedEvent with property values and ID 1`] = `
{
  "context": "test-context",
  "description": "Grocery category",
  "id": Any<Number>,
  "name": "Grocery",
  "parentId": null,
}
`;

exports[`Category create Success should apply CategoryCreatedEvent with property values and no ID 1`] = `
{
  "context": "test-context",
  "description": "Grocery category",
  "id": undefined,
  "name": "Grocery",
  "parentId": null,
}
`;

exports[`Category update should apply CategoryUpdatedEvent with properties to be updated 1`] = `
CategoryUpdatedEvent {
  "context": "new-context",
  "description": "Grocery category",
  "id": undefined,
  "name": "Grocery",
  "parentId": null,
}
`;
